lib/pos_blockchain.go

```diff
@@ -374,17 +366,144 @@ func (bc *Blockchain) getLineageFromCommittedTip(desoBlock *MsgDeSoBlock) ([]*Bl
 	return ancestors, nil
 }
 
-// addBlockToBlockIndex adds the block to the block index and uncommitted blocks map.
-func (bc *Blockchain) addBlockToBlockIndex(desoBlock *MsgDeSoBlock, blockStatus BlockStatus) error {
-	hash, err := desoBlock.Hash()
+// getOrCreateBlockNodeFromBlockIndex returns the block node from the block index if it exists.
+// Otherwise, it creates a new block node and adds it to the block index.
+func (bc *Blockchain) getOrCreateBlockNodeFromBlockIndex(desoBlock *MsgDeSoBlock) (*BlockNode, error) {
```

Comment:
nit: may as well refer to blocks as "block" and block nodes as "blockNode" for clarity
```suggestion
func (bc *Blockchain) getOrCreateBlockNodeFromBlockIndex(block *MsgDeSoBlock) (*BlockNode, error) {
```